### ISSUE TITLE
[B3.2] Add CosineAnnealingLR and ReduceLROnPlateau schedulers

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2304]
## Overview
Add advanced learning rate schedulers to shared/autograd/schedulers.mojo.

## Location
- `shared/autograd/schedulers.mojo` (created in B3.1)

### CosineAnnealingLR
Cosine annealing with warm restarts:
```
lr = lr_min + 0.5 * (lr_max - lr_min) * (1 + cos(π * T_cur / T_max))
```

### ReduceLROnPlateau
Reduce learning rate when metric stops improving:
- Monitor a metric (e.g., validation loss)
- Reduce lr by factor when no improvement for patience epochs

## Deliverables
- [ ] Implement CosineAnnealingLR scheduler:
  - T_max: maximum number of epochs
  - eta_min: minimum learning rate
- [ ] Implement ReduceLROnPlateau scheduler:
  - mode: 'min' or 'max'
  - factor: lr reduction factor
  - patience: epochs to wait
  - threshold: threshold for measuring improvement
- [ ] Add tests for both schedulers

## Success Criteria
- Schedulers work with training loops
- Correct learning rate behavior
- CI passes

## Dependencies
- B3.1 (schedulers.mojo base)

## Track
Track B: Missing Features (B3.2 of 8)
